jsinterop-ts-defs-impl/src/main/java/com/vertispan/tsdefs/impl/builders/TsElement.java

```diff
@@ -756,6 +756,7 @@ public boolean requiresProtectedConstructor() {
           element.getEnclosedElements().stream()
               .map(enclosedElement -> TsElement.of(enclosedElement, env))
               .filter(TsElement::isConstructor)
+              .filter(tsElement -> !tsElement.isPrivate())
```

Comment:
as long as you're in here, wouldnt it make more sense to also move the `isIgnored` filter here - functionally, we filter out ignored and private constructors in the same way (and this list isn't used anywhere else).

Don't even need the list property then, just a findAny().isPresent() to assign to allIgnored?